### PR TITLE
support for external modules

### PR DESCRIPTION
**Pull request by jguddas**
I'm not much of an expert but i gave moving the modules out of the main repo a try and it seems to be working :smile:.

### CODE REVIEW

**Comment by jguddas** [comment]
@ericbn what do you think about this?

**Comment by braineo** [comment]
I think this feature is cool, managing submodule is not an advantage of git.

File: tools/zim_update [hunk 1]
```diff
@@ -2,10 +2,35 @@
...
 #
 
+echo "\nUpdating Zim"
 cd ${ZIM_HOME}
 
 # this is the cleanest way I know how to update a repository
 git remote update -p
 git merge --ff-only @\{u\}
 # and update the submodules
 git submodule update --init --recursive
```
Comment (kaznovac) [inline tools/zim_update]: this is unnecessary as there are no submodules left :)

Comment (jguddas) [reply tools/zim_update]: this is so people can still manage their modules using submodules if they so chose to.

**Comment by ericbn** [comment]
@jguddas, I've implemented this in the new develop branch of Zim. See #323 and see https://github.com/zimfw/zimfw/tree/develop

**Comment by ericbn** [comment]
This is done with the release of Zim 1.0.0.

**Pull request closed without merging**